### PR TITLE
Add phylopic_key_glyph

### LINKED ISSUE #57: custom legend icons for geom_phylopic
@LewisAJones requests that legends related to `geom_phylopic` should use the silhouettes instead of points

### PR DESCRIPTION
**Pull request by willgearty**
This adds a custom key glyph function that can be used with `geom_phylopic` to specify silhouettes for the legend.
image

Resolves #57.

### CODE REVIEW

File: R/geom_phylopic.R [hunk 1]
```diff
@@ -182,6 +189,109 @@ GeomPhylopic <- ggproto("GeomPhylopic", Geom,
...
+#' @param uuid \code{character}. A valid uuid for a PhyloPic silhouette (such as
+#'   that returned by [get_uuid()] or [pick_phylopic()]).
+#' @details One (and only one) of `img`, `name`, or `uuid` must be specified.
+#'
+#'   This argument may also be a vector/list of valid values if multiple
+#'   silhouettes should be used as key glyphs. In this case, the specified
+#'   silhouettes will be used as ordered as key glyphs one by one, with
+#'   recycling as necessary.
+#'
+#'   Note that the relative sizes of the silhouettes are fixed. Therefore, this
```
Comment (LewisAJones) [inline R/geom_phylopic.R:211]: No straightforward way of addressing this?

**Review by LewisAJones — CHANGES_REQUESTED** [review]
This works really well @willgearty. I know it gave you some pain, so great job. A few things we might want to consider:

- [ ] Ensure this works with the updated aesthetics for `geom_phylopic` (use of colour and fill) and outline feature.
- [ ] Can we make this work with the size aesthetic as well?
- [ ] Do we want an option for creating legends for `add_phylopic` and `add_phylopic_base`?

**Comment by willgearty** [comment]
Note: I still need to figure out why one of the tests isn't making the correct snapshot.

**Comment by willgearty** [comment]
With the most recent commit, looks like this "just works" with the new fill/color split.

File: R/geom_phylopic.R (hunk 1 above)
Comment (willgearty) [reply R/geom_phylopic.R:243]: I'll give it another go, but the last time I tried it made the silhouettes REALLY big.

Comment (LewisAJones) [reply R/geom_phylopic.R:243]: OK, have a little bit of a play, but don't sweat over it for now. Colour/fill is likely to be the main aesthetics we need to get right.

**Review by LewisAJones — APPROVED** [review]
Hi @willgearty, this is working well! Last thing from me is whether we expect `color` defined outside of the aesthetics to also work when multiple values are defined? I think we should be encouraging it to be defined in the aesthetics, but just in case, I thought I'd mention it.

```
library(ggplot2)
df <- data.frame(x = c(2, 4), y = c(10, 20),
                 name = c("Felis silvestris catus", "Odobenus rosmarus"))
# Works with outline colour
ggplot(df) +
  geom_phylopic(aes(x = x, y = y, name = name, fill = name), color = c("blue"), 
                size = 10,
                show.legend = TRUE,
                key_glyph = phylopic_key_glyph(name =
                                                 c("Felis silvestris catus",
                                                   "Odobenus rosmarus"))) +
  coord_cartesian(xlim = c(1,6), ylim = c(5, 30))
# Doesn't work with outline colour (defaults to black)
ggplot(df) +
  geom_phylopic(aes(x = x, y = y, name = name, fill = name), color = c("blue", "red"), 
                size = 10,
                show.legend = TRUE,
                key_glyph = phylopic_key_glyph(name =
                                                 c("Felis silvestris catus",
                                                   "Odobenus rosmarus"))) +
  coord_cartesian(xlim = c(1,6), ylim = c(5, 30))
```

**Comment by willgearty** [comment]
This is the expected behavior for a ggplot legend. For example:
```r
ggplot(df) +
    geom_point(aes(x = x, y = y, fill = name), color = c("blue", "red"), size = 10, shape = 21) +
    coord_cartesian(xlim = c(1,6), ylim = c(5, 30))
```

**Comment by willgearty** [comment]
OK, your two other requests/comments have been moved to separate issues for future development. I'll merge this once the checks pass!

**Pull request merged**